kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java

```diff
@@ -286,10 +259,12 @@ private Subscription createSubscriptionViaMcts(Subscriber subscriber, Subscripti
             }
         } else if (subscriber.getLastMenstrualPeriod() != null && subscriber.getDateOfBirth() == null &&
```

Comment:
Why check for DOB here and not check for LMP on line 247?

Or

The logic here seems to be complex and I'm having a hard time understanding if it's right or wrong.